kie-server-parent/kie-server-tests/kie-server-integ-tests-dmn/src/test/java/org/kie/server/integrationtests/dmn/DMNBkmJavaModelTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.integrationtests.dmn;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Map;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.kie.api.KieServices;
+import org.kie.dmn.api.core.DMNContext;
+import org.kie.dmn.api.core.DMNResult;
+import org.kie.server.api.model.KieServiceResponse.ResponseType;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.integrationtests.shared.KieServerDeployer;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+public class DMNBkmJavaModelTest
+        extends DMNKieServerBaseIntegrationTest {
+
+    private static final ReleaseId kjar1 = new ReleaseId("org.kie.server.testing", "bkm-java-model", "1.0.0.Final");
+
+    private static final String CONTAINER_1_ID = "bkm-java-model";
+    private static final String CONTAINER_1_ALIAS = "bkm-java";
+
+    @BeforeClass
+    public static void deployArtifacts() {
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/bkm-java-model").getFile());
+
+        kieContainer = KieServices.Factory.get().newKieContainer(kjar1);
+        createContainer(CONTAINER_1_ID, kjar1, CONTAINER_1_ALIAS);
+    }
+
+    @Override
+    protected void addExtraCustomClasses(Map<String, Class<?>> extraClasses)
+            throws Exception {
+
+        // no extra classes.
+    }
+
+    @Test
+    public void test_evaluateAll() {
+        DMNContext dmnContext = dmnClient.newContext();
+        dmnContext.set("Values", Arrays.asList(new BigDecimal(1), new BigDecimal(2), new BigDecimal(3)));
```

Comment:
I have a problem marshalling `ArrayList` only in the case of JAXB.
I tried substituting with explict ArrayList, arrays, etc. but no dice, I always get the following problem:

```
13:49:20.068 [Acme.Utils.ThreadPool(1)-PooledThread: Acme.Serve.Serve$ServeConnection@5bc4a2bd] DEBUG o.k.s.s.d.ModelEvaluatorServiceBase - Will deserialize payload: <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<dmn-evaluation-context>
    <dmn-context xsi:type="jaxbListWrapper" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
        <type>MAP</type>
        <element xsi:type="jaxbStringObjectPair" key="Values">
            <value xsi:type="jaxbListWrapper">
                <type>LIST</type>
                <element xsi:type="xs:decimal" xmlns:xs="http://www.w3.org/2001/XMLSchema">1</element>
                <element xsi:type="xs:decimal" xmlns:xs="http://www.w3.org/2001/XMLSchema">2</element>
                <element xsi:type="xs:decimal" xmlns:xs="http://www.w3.org/2001/XMLSchema">3</element>
            </value>
        </element>
    </dmn-context>
</dmn-evaluation-context>

java.lang.NullPointerException
	at java.util.EnumMap.<init>(EnumMap.java:139)
	at com.sun.xml.bind.v2.model.impl.RuntimeEnumLeafInfoImpl.<init>(RuntimeEnumLeafInfoImpl.java:87)
	at com.sun.xml.bind.v2.model.impl.RuntimeModelBuilder.createEnumLeafInfo(RuntimeModelBuilder.java:108)
	at com.sun.xml.bind.v2.model.impl.RuntimeModelBuilder.createEnumLeafInfo(RuntimeModelBuilder.java:84)
	at com.sun.xml.bind.v2.model.impl.ModelBuilder.getClassInfo(ModelBuilder.java:242)
	at com.sun.xml.bind.v2.model.impl.RuntimeModelBuilder.getClassInfo(RuntimeModelBuilder.java:103)
	at com.sun.xml.bind.v2.model.impl.RuntimeModelBuilder.getClassInfo(RuntimeModelBuilder.java:84)
	at com.sun.xml.bind.v2.model.impl.ModelBuilder.getClassInfo(ModelBuilder.java:227)
	at com.sun.xml.bind.v2.model.impl.RuntimeModelBuilder.getClassInfo(RuntimeModelBuilder.java:98)
	at com.sun.xml.bind.v2.model.impl.RuntimeModelBuilder.getClassInfo(RuntimeModelBuilder.java:84)
	at com.sun.xml.bind.v2.model.impl.ModelBuilder.getTypeInfo(ModelBuilder.java:375)
	at com.sun.xml.bind.v2.model.impl.ModelBuilder.getTypeInfo(ModelBuilder.java:390)
	at com.sun.xml.bind.v2.runtime.JAXBContextImpl.getTypeInfoSet(JAXBContextImpl.java:454)
	at com.sun.xml.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:292)
	at com.sun.xml.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:139)
	at com.sun.xml.bind.v2.runtime.JAXBContextImpl$JAXBContextBuilder.build(JAXBContextImpl.java:1138)
	at com.sun.xml.bind.v2.ContextFactory.createContext(ContextFactory.java:162)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:247)
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:234)
	at javax.xml.bind.ContextFinder.find(ContextFinder.java:441)
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:641)
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:584)
	at org.kie.server.api.marshalling.jaxb.JaxbMarshaller.buildMarshaller(JaxbMarshaller.java:394)
	at org.kie.server.api.marshalling.jaxb.JaxbMarshaller.<init>(JaxbMarshaller.java:380)
	at org.kie.server.api.marshalling.BaseMarshallerBuilder.build(BaseMarshallerBuilder.java:40)
	at org.kie.server.api.marshalling.MarshallerFactory.getMarshaller(MarshallerFactory.java:52)
	at org.kie.server.services.impl.KieContainerInstanceImpl.getMarshaller(KieContainerInstanceImpl.java:175)
	at org.kie.server.services.impl.marshal.MarshallerHelper.unmarshal(MarshallerHelper.java:102)
	at org.kie.server.services.impl.marshal.MarshallerHelper.unmarshal(MarshallerHelper.java:87)
	at org.kie.server.services.dmn.ModelEvaluatorServiceBase.evaluateDecisions(ModelEvaluatorServiceBase.java:153)
	at org.kie.server.remote.rest.dmn.ModelEvaluatorResource.evaluateDecisions(ModelEvaluatorResource.java:108)
```

While JSON and XStream serialization looks to work just fine. Any advice please?

Reply:
since this fails at JAXB Context init time I suspect this is related to some of the dependencies included in the project. So it might be as mentioned in the other comment - use provided scope for added dependency or try turning on class filtering:
```
-Dorg.drools.server.filter.classes=true
```